test/e2e-go/features/stateproofs/stateproofs_test.go

```diff
@@ -226,6 +228,80 @@ func TestStateProofOverlappingKeys(t *testing.T) {
 	r.Equalf(int(consensusParams.StateProofInterval*expectedNumberOfStateProofs), int(lastStateProofBlock.Round()), "the expected last state proof block wasn't the one that was observed")
 }
 
+func TestBlkMerkleProof(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	r := require.New(fixtures.SynchronizedTest(t))
+
+	configurableConsensus := make(config.ConsensusProtocols)
+	consensusVersion := protocol.ConsensusVersion("test-fast-stateproofs")
+	consensusParams := config.Consensus[protocol.ConsensusCurrentVersion]
+	consensusParams.StateProofInterval = 16
+	consensusParams.StateProofTopVoters = 1024
+	consensusParams.StateProofVotersLookback = 2
+	consensusParams.StateProofWeightThreshold = (1 << 32) * 30 / 100
+	consensusParams.StateProofStrengthTarget = 256
+	consensusParams.StateProofRecoveryInterval = 6
+	consensusParams.EnableStateProofKeyregCheck = true
+	consensusParams.AgreementFilterTimeout = 1500 * time.Millisecond
+	consensusParams.AgreementFilterTimeoutPeriod0 = 1500 * time.Millisecond
+	configurableConsensus[consensusVersion] = consensusParams
+
+	var fixture fixtures.RestClientFixture
+	fixture.SetConsensus(configurableConsensus)
+	fixture.Setup(t, filepath.Join("nettemplates", "StateProof.json"))
+	defer fixture.Shutdown()
+
+	libgoalClient := fixture.LibGoalClient
+
+	restClient, err := fixture.NC.AlgodClient()
+	r.NoError(err)
+
+	var startRound = uint64(1)
+	var nextStateProofRound = uint64(0)
+	var firstStateProofRound = 2 * consensusParams.StateProofInterval
+
+	t.Logf("Waiting for state proof to be committed.")
+
+	for rnd := startRound; nextStateProofRound <= firstStateProofRound; rnd++ {
```

Comment:
What do you think about refactoring here? I think there might be some common code with previous tests. Not sure it's worth the effort. @algoidan